File: core/src/main/python/wlsdeploy/util/exit_code.py
"""
Copyright (c) 2022, Oracle Corporation and/or its affiliates.
Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.

Standard exit codes for command-line utilities.
"""

from oracle.weblogic.deploy.util import ExitCode as JExitCode

class ExitCode(object):
    """
    Standard exit codes for command-line utilities.
    """

    OK                         = JExitCode.OK
    WARNING                    = JExitCode.WARNING
    ERROR                      = JExitCode.ERROR

    ARG_VALIDATION_ERROR       = JExitCode.ARG_VALIDATION_ERROR
    USAGE_ERROR                = JExitCode.USAGE_ERROR

    HELP                       = JExitCode.HELP
    DEPRECATION                = JExitCode.DEPRECATION
    RESTART_REQUIRED           = JExitCode.RESTART_REQUIRED
    CANCEL_CHANGES_IF_RESTART  = JExitCode.CANCEL_CHANGES_IF_RESTART

File: core/src/main/java/oracle/weblogic/deploy/util/ExitCode.java
/*
 * Copyright (c) 2023, Oracle and/or its affiliates.
 * Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
 */
package oracle.weblogic.deploy.util;

public class ExitCode {
    public static final int OK = 0;
    public static final int WARNING = 1;
    public static final int ERROR = 2;
    public static final int ARG_VALIDATION_ERROR = 98;
    public static final int USAGE_ERROR = 99;
    public static final int HELP = 100;
    public static final int DEPRECATION = 101;
    public static final int RESTART_REQUIRED = 103;
    public static final int CANCEL_CHANGES_IF_RESTART = 104;

    private ExitCode() {
        // hide default constructor
    }
}
